include/eve/arch/cpu/logical.hpp

```diff
@@ -44,7 +44,13 @@ namespace eve
     //==============================================================================================
     // Assignment
     //==============================================================================================
-    EVE_FORCEINLINE constexpr logical &operator=(bool v) noexcept
+    EVE_FORCEINLINE constexpr logical &operator=(logical v) & noexcept
+    {
+      value_ = v.value_;
+      return *this;
+    }
+
+    EVE_FORCEINLINE constexpr logical &operator=(bool v) & noexcept
```

Comment:
This makes no sense, sorry.

I mean - one should not be able to assign to a temporary in any case.
This is not a good fix for it.

Do you just return a `logical` by value?

I would suggest to drop `operator[]` all together or return a proxy object.

But this is not very good I think.

Reply:
I added some references. If you're OK as per our Slack discussion, I'll proceed to the merge

Reply:
Merging